internal/ec2/securitygroups.go

```diff
@@ -2,31 +2,52 @@ package ec2
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/Method-Security/methodaws/internal/sts"
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
 )
 
-// EnumerateSecurityGroupsForRegion lists all of the security groups available to the caller alongside any non-fatal errors that
-// occurred during the execution of the `methodaws securitygroup enumerate` subcommand.
+// EnumerateSecurityGroups lists all of the security groups available to the caller across multiple regions
+// alongside any non-fatal errors that occurred during the execution of the `methodaws securitygroup enumerate` subcommand.
 // If vpcID is not nil, it will only return security groups associated with that VPC.
-func EnumerateSecurityGroupsForRegion(ctx context.Context, cfg aws.Config, vpcID *string, region string) (SecurityGroupReport, error) {
-	cfg.Region = region
+func EnumerateSecurityGroups(ctx context.Context, cfg aws.Config, vpcID *string, regions []string) (SecurityGroupReport, error) {
+	var allSecurityGroups []types.SecurityGroup
+	var allErrors []string
+	var accountID string
+
+	if len(regions) > 0 {
+		id, err := sts.GetAccountID(ctx, cfg)
+		if err != nil {
+			allErrors = append(allErrors, fmt.Sprintf("Error getting account ID: %v", err))
```

Comment:
Realistically we end here, do we not? We need the account ID to send a valid signal